lib/cn/kvset_split.c

```diff
@@ -242,6 +245,13 @@ kblock_split(
 
     INVARIANT(kbd && split_key && kblks);
 
+    /* Readahead source kblock regions to improve read latency when iterating over its keys
+     * inside kblock_copy_range().
+     */
+    kbr_madvise_kmd(kbd->kd_mbd, kbd->kd_wbd, MADV_WILLNEED);
+    kbr_madvise_wbt_int_nodes(kbd->kd_mbd, kbd->kd_wbd, MADV_WILLNEED);
+    kbr_madvise_wbt_leaf_nodes(kbd->kd_mbd, kbd->kd_wbd, MADV_WILLNEED);
```

Comment:
Are both int and leaf wbt's populated?  I forget, seems like leaf nodes reference the next level nodes?

Reply:
You're right.. I'm going to try without the madvise for the internal nodes.

Reply:
If you're doing lookups you need internal and leaf nodes.  If you're iterating you only need leaf nodes.